grub-core/net/bootp.c

```diff
@@ -31,6 +31,8 @@
 #define GRUB_NET_BOOTP_ARCH 0x0007
 #elif defined(GRUB_MACHINE_EFI) && defined(__aarch64__)
 #define GRUB_NET_BOOTP_ARCH 0x000B
+#elif defined(__powerpc__)
```

Comment:
For starters we should use `defined(GRUB_MACHINE_IEEE1275) && defined(__powerpc__)` but that still doesn't seem accurate since there is also values defined for "PowerPC Open Firmware" and "PowerPC ePAPR" in addition to "POWER OPAL v3" in http://www.iana.org/assignments/dhcpv6-parameters/dhcpv6-parameters.xhtml#processor-architecture I presume all of those different firmware types use the same "powerpc_ieee1275" grub so we can't really know the correct value at compile time. Is that correct? Those different values may not be applicable for CoreOS but it'll be a concern if/when we get around to contributing this code upstream.

All that said, upon review commit fbf65674a471d12e43f4050086969f417e00891f that added this in the first place is buggy and doesn't actually allocate any memory for this option it is adding to the DHCP packet... @mischief @mjg59